tests/foreman/cli/test_discoveryrule.py

```diff
@@ -222,8 +222,8 @@ def test_positive_create_with_priority(self, discoveryrule_factory):
 
         :CaseImportance: Critical
         """
-        available = set(range(1, 1000)) - {Box(r).priority for r in DiscoveryRule.list()}
-        rule_priority = random.sample(available, 1)
+        available = set(range(1, 1000)) - {int(Box(r).priority) for r in DiscoveryRule.list()}
+        rule_priority = random.sample(sorted(available), 1)
```

Comment:
Why would we sort the set?

Reply:
I just followed the error`E   TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).` :)

Reply:
Hmmm ok I thought this was a different error. I used to sometimes get a BoxKeyError that would say the rule coming out of `DiscoveryRule.list()` had no priority attribute.

Reply:
Good catch with the error @Gauravtalreja1